godtossing/Pos/Binary/GodTossing/Core.hs

```diff
@@ -163,3 +99,35 @@ decodeCommitments = do
     unless (allDistinct (map (view _1) comms :: [PublicKey])) $
         fail "decodeCommitments: two commitments have the same signing key"
     pure (mkCommitmentsMap comms)
+
+----------------------------------------------------------------------------
+-- TH-generated instances go to the end of the file
+----------------------------------------------------------------------------
+
+deriveSimpleBi ''GtPayload [
```

Comment:
:facepalm:

Reply:
tbh I forgot it can generate instances for datatype for several constructors, thx for reminder